app-games/src/main/java/com/aptoide/android/aptoidegames/apkfy/analytics/ApkfyManagerProbe.kt

```diff
@@ -4,67 +4,67 @@ import cm.aptoide.pt.feature_apkfy.domain.ApkfyManager
 import cm.aptoide.pt.feature_apkfy.domain.ApkfyModel
 import cm.aptoide.pt.feature_campaigns.AptoideMMPCampaign
 import com.aptoide.android.aptoidegames.IdsRepository
-import com.aptoide.android.aptoidegames.analytics.BIAnalytics
+import com.google.gson.Gson
+import kotlinx.coroutines.delay
+import retrofit2.HttpException
 
 class ApkfyManagerProbe(
   private val apkfyManager: ApkfyManager,
-  private val biAnalytics: BIAnalytics,
+  private val apkfyAnalytics: ApkfyAnalytics,
   private val idsRepository: IdsRepository,
 ) : ApkfyManager {
 
   companion object {
-    private const val UTM_PROPERTY_NO_APKFY = "NO_APKFY"
-    private const val UTM_PROPERTY_APKFY_WITHOUT_UTMS = "APKFY_BUT_NO_UTM"
     private const val GUEST_UID_KEY = "GUEST_UID"
   }
 
   override suspend fun getApkfy(): ApkfyModel? {
     if (idsRepository.getId(GUEST_UID_KEY).isEmpty()) {
-      return apkfyManager.getApkfy()
-        ?.also(::setApkfyUTMProperties)
-        ?.also {
-          idsRepository.saveId(GUEST_UID_KEY, it.guestUid)
-        }.also {
-          // TODO: improve this logic
-          AptoideMMPCampaign.guestUID = it?.guestUid ?: idsRepository.getId(GUEST_UID_KEY)
+      var isRetry = false
+      var apkfyModel: ApkfyModel? = null
+      var attempts = 0
+
+      while (attempts < 3) { //Apkfy call repeated at most 3 times, to make sure there is no apkfy app associated
+        try {
+          apkfyModel = apkfyManager.getApkfy()
```

Comment:
I don't think this repeat/retry logic belongs to the probe.
Let's not mix the business logic of repeat/retries with the analytics logic of the probe 🤔